ec_setup.pl

```diff
@@ -65,9 +69,53 @@
 
 
 $logfile .= $dslReponse->findnodes_as_string("/");
-
 my $errorMessage = $commander->getError();
 
+if ( !$errorMessage ) {
+    # This is here because we cannot do publishArtifactVersion in dsl today
+    # delete artifact if it exists first
+    $commander->deleteArtifactVersion("com.electriccloud:PluginWizard-Grapes:1.0.1");
+
+    my $dependenciesProperty = '/projects/@PLUGIN_NAME@/ec_groovyDependencies';
+    my $base64 = $commander->getProperty($dependenciesProperty)->findvalue('//value')->string_value;
+    my $binary = decode_base64($base64);
+    my ($tempFh, $tempFilename) = tempfile(CLEANUP => 1);
+    binmode($tempFh);
+    print $tempFh $binary;
+    close $tempFh;
+
+    my ($tempDir) = tempdir(CLEANUP => 1);
+    my $zip = Archive::Zip->new();
+    unless($zip->read($tempFilename) == Archive::Zip::AZ_OK()) {
+      die "Cannot read .zip dependencies: $!";
+    }
+    $zip->extractTree("", File::Spec->catfile($tempDir, ''));
+    warn "Dependencies: $tempDir\n";
+
+    if ( $promoteAction eq "promote" ) {
+
+        #publish jars to the repo server if the plugin project was created successfully
+        my $am = new ElectricCommander::ArtifactManagement($commander);
```

Comment:
The README has steps for using the PluginWizard as a starting point for creating a new plugin. If the plugin now requires the grape dependency libraries then the artifact name to use for the grape dependencies should use the user assigned plugin name instead of 'PluginWizard' so that the user can still follow the steps in README to create a new working plugin.

Reply:
Changed ec_setup, so it would not publish the artifact if there is no dependencies found.